packages/harmony/src/components/text/Text/Text.tsx

```diff
@@ -51,21 +57,26 @@ export const Text = forwardRef(
         lineHeight: typography.lineHeight[variantConfig.lineHeight[size]],
         // @ts-ignore
         fontWeight: typography.weight[variantConfig.fontWeight[strength]],
-        ...('css' in variantConfig && variantConfig.css),
-        ...(shadow && {
-          textShadow: typography.shadow[shadow]
-        }),
-        textAlign
+        ...('css' in variantConfig && variantConfig.css)
+      }),
+      ...(shadow && {
+        textShadow: typography.shadow[shadow]
+      }),
+      textAlign,
+      ...(ellipses && {
+        overflow: 'hidden',
+        textOverflow: 'ellipsis',
+        whiteSpace: 'nowrap'
       })
     }
 
     // @ts-ignore
     const variantTag = variant && variantTagMap[variant]?.[size]
 
-    const Tag: ElementType = asChild ? Slot : tag ?? variantTag ?? 'p'
+    const Tag: ElementType = asChild ? Slot : tag ?? variantTag ?? 'span'
```

Comment:
curious about the change here. i feel like `p` should be the default tag, but seems like inline styles might be needed more commonly?

Reply:
so what i saw was that we were constantly getting "invalid dom nesting" cause a div would be inside a p tag, or a p tag would be in another p tag. what you think the move is?

Reply:
ah right, does it not complain about divs inside of span elements? if not then this seems better. i think rules around not putting specific things within text components would be too cumbersome